VCFv4.5.draft.tex

```diff
@@ -578,6 +585,38 @@ \subsubsection{Genotype fields}
     \end{itemize}
 
   \item HQ (Integer): Haplotype qualities, two comma separated phred qualities.
+  \item LAA is a sorted list of $n$ distinct integers, where $0 \le n \le \left|\mathrm{ALT}\right|$, giving the indices of the alleles that are observed in the sample.
+  In callsets with many samples, sites may grow to include numerous alternate alleles at the same POS.
+  Usually, few of these alleles are actually observed in any one sample, but each genotype must supply fields like PL and AD for all of the alleles---a very inefficient representation as PL's size is quadratic in the allele count.
+  Similarly, in rare sites, which can be the bulk of the sites, the vast majority of the samples are reference.
+  To prevent this growth in VCF size, one can choose to specify the genotype, allele depth and the genotype likelihood against a subset of ``Local Alleles''.
+  LAA is the strictly increasing index into REF and ALT, pointing out the alleles that are actually in-play for that sample.
+  0 indicates the REF allele and should always be included with the subsequent values being 1-based indexes into ALT.
+  LAD is the depth of the local alleles,
+  LPL is subset of the PL array that pertains to the alleles that are referred to by LAA,
```

Comment:
Similarly here.  Eg in a classic world we may have:

```
GT:PL:AD        0/1:187,0,28:2,7
```

In local world, I'm struggling and the above explanations seem to be stating something wrong.  With `LAA 0` (one ALT), LPL would refer to both REF and ALT#0, not ALT#0 only.  Hence "pertains to the alleles that are referred to by LAA" should be "pertains to the alleles that are referred to by LAA and REF".

I assume similarly true for LAD which claims to be "depth of the local alleles".  Isn't it "depth of REF and the local alleles"?

Basically, the only way I can see this working is if all the local versions are REF + whatever is listed in LAA.  So I would argue that perhaps it *is* the right thing to include REF in LAA then all of the above statements start becoming true once more.

Reply:
The text of this proposal is incorrect. It says LAA but describes LA instead.

LAA is a list of increasing, 1-based indices into ALT alleles. The original pull request states this correctly https://github.com/samtools/hts-specs/pull/434.

Reply:
Ah I see.  Thanks.  That explains my confusion over numbering then.  I hadn't realised it had changed LAA this way.  It's a bad idea.  EIther we should keep LAA as it was or switch to LA, but using LA semantics in LAA will just break a whole bunch of existing software and make it impossible to parse existing files.